src/main/java/org/folio/rest/impl/UploadBatchVoucherExportHelper.java

```diff
@@ -127,7 +128,13 @@ private void updateHolderWithFileFormat(BatchVoucherUploadHolder uploadHolder) {
 
   private Future<Void> updateHolderWithCredentials(BatchVoucherUploadHolder uploadHolder) {
     return batchVoucherExportConfigService.getExportConfigCredentials(uploadHolder.getExportConfig().getId(), requestContext)
-      .onSuccess(uploadHolder::setCredentials)
+      .compose(credentials -> {
+        if (StringUtils.isBlank(credentials.getUsername()) || StringUtils.isBlank(credentials.getPassword())) {
+          throw new HttpException(404, CREDENTIALS_NOT_FOUND);
+        }
+        uploadHolder.setCredentials(credentials);
+        return Future.succeededFuture(null);
+      })
       .recover(t -> {
         throw new HttpException(404, CREDENTIALS_NOT_FOUND);
```

Comment:
I think we can improve more here, lets imagine that we throw an exception with another error code in compose section, in this case recover will always rewrite original exception.
As solution we can add .recover after .getExportConfigCredentials and at the end use compose where checking credentials

Reply:
This validation was removed, because empty credentials are no longer saved to db

Reply:
https://github.com/folio-org/mod-invoice/pull/471/commits/27b3170dc6869c4b481ef0205ca0d22a2328a293#diff-bf1e298570cf67f2736194d7d458d26e8cb6c2933ba1d9b9e352f520f920bb05L58-R73